src/adiimu_ros_publisher.cpp

```diff
@@ -58,29 +58,43 @@ void AdiImuRosPublisher::run()
-        RCLCPP_INFO(rclcpp::get_logger("rclcpp_adiimu"), "Publishing adi imu gyroscope data: '%f' '%f' '%f'",
-                    m_message.gyro.x, m_message.gyro.y, m_message.gyro.z);
-        RCLCPP_INFO(rclcpp::get_logger("rclcpp_adiimu"), "Publishing adi imu delta velocity data: '%f' '%f' '%f'",
-                    m_message.delta_vel.x, m_message.delta_vel.y, m_message.delta_vel.z);
-        RCLCPP_INFO(rclcpp::get_logger("rclcpp_adiimu"), "Publishing adi imu delta angle data: '%f' '%f' '%f'",
-                    m_message.delta_angle.x, m_message.delta_angle.y, m_message.delta_angle.z);
-        RCLCPP_INFO(rclcpp::get_logger("rclcpp_adiimu"), "Publishing adi imu temperature data: '%f' ",
-                    m_message.temp);
-        RCLCPP_INFO(rclcpp::get_logger("rclcpp_adiimu"), "Publishing adi imu time %d us ",
-                    std::chrono::duration_cast<std::chrono::microseconds>(done-started).count());
+        if(operation_mode == SAMPLING_MODE)
+        {
+            auto started = std::chrono::high_resolution_clock::now();
+            bool success = false;
+            m_message = m_dataProvider->getData(success);
+            auto done = std::chrono::high_resolution_clock::now();
 
-        if(success)
-            m_publisher->publish(m_message);
-        count++;
-        //rclcpp::spin_some(m_node);
-        //loopRate.sleep();
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_adiimu"), "Publishing adi imu acceleration data: '%f' '%f' '%f'",
+                        m_message.accel.x, m_message.accel.y, m_message.accel.z);
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_adiimu"), "Publishing adi imu gyroscope data: '%f' '%f' '%f'",
+                        m_message.gyro.x, m_message.gyro.y, m_message.gyro.z);
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_adiimu"), "Publishing adi imu delta velocity data: '%f' '%f' '%f'",
+                        m_message.delta_vel.x, m_message.delta_vel.y, m_message.delta_vel.z);
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_adiimu"), "Publishing adi imu delta angle data: '%f' '%f' '%f'",
+                        m_message.delta_angle.x, m_message.delta_angle.y, m_message.delta_angle.z);
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_adiimu"), "Publishing adi imu temperature data: '%f' ",
+                        m_message.temp);
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_adiimu"), "Publishing adi imu time %d us ",
+                        std::chrono::duration_cast<std::chrono::microseconds>(done-started).count());
+
+            if(success)
+                m_publisher->publish(m_message);
+            count++;
+            //rclcpp::spin_some(m_node);
+            //loopRate.sleep();
+        }
+        else
+            if(operation_mode == STOP_LOOP)
+            {
+                break;
+            }
+            else
+            {
+                loopRate.sleep();
+            }
```

Comment:
braces are not necessary in this case and else if should go on the line of code.
braces are needed only when more than one instruction is present in if or else conditions.